cmake/compiler_config.cmake

```diff
@@ -38,6 +37,19 @@ set(MSVC_FLAGS
     /wd4180
 )
 
+# The /Zc:inline option strips out the "arch_ctor_<name>" symbols used for
+# library initialization by ARCH_CONSTRUCTOR starting in Visual Studio 2019,
+# causing release builds to fail. Disable the option for this and later
+# versions.
+#
+# For more details, see:
+# https://developercommunity.visualstudio.com/content/problem/914943/zcinline-removes-extern-symbols-inside-anonymous-n.html
+if (MSVC_VERSION GREATER_EQUAL 1920)
+    set(MSVC_FLAGS "${MSVC_FLAGS} /Zc:inline-")
+else()
+    set(MSVC_FLAGS "${MSVC_FLAGS} /Zc:inline")
+endif()
+
```

Comment:
Nit-pick: You could also use regular expression to have you if/else logic in one line and keep it inside the MSVC_FLAGS list.

`$<IF:$<VERSION_GREATER_EQUAL:${MSVC_VERSION},1920>,/Zc:inline-,/Zc:inline>`

Reply:
That is a better syntax indeed. Thanks!